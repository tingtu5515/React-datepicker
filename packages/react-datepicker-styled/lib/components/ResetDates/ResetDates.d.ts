interface onResetDatesProps {
    onResetDates(): void;
}
declare function ResetDates({ onResetDates }: onResetDatesProps): JSX.Element;
export default ResetDates;
