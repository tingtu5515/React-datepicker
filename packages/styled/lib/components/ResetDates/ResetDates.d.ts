interface onResetDatesProps {
  onResetDates(): void
  text: string
}
declare function ResetDates({onResetDates, text}: onResetDatesProps): JSX.Element
export default ResetDates
