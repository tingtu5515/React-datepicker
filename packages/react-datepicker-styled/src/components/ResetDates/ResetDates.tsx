import React from 'react'
import RedoIcon from '../../icons/RedoIcon'
import Text from '../Text'
import Flex from '../Flex'

interface onResetDatesProps {
  onResetDates(): void
}

function ResetDates({onResetDates}: onResetDatesProps) {
  return (
    <Flex role="button" onClick={onResetDates} alignItems="center">
      <RedoIcon height="14px" width="14px" color="58595B" />
      <Text
        ml="8px"
        mt="1px"
        lineHeight={1.18}
        fontFamily="Montserrat"
        fontSize="11px"
        color="#343132"
      >
        Reset dates
      </Text>
    </Flex>
  )
}

export default ResetDates
