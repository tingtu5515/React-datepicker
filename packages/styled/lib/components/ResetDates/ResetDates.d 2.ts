interface ResetDatesProps {
  onResetDates(): void
  text: string
}
declare function ResetDates({onResetDates, text}: ResetDatesProps): JSX.Element
export default ResetDates
