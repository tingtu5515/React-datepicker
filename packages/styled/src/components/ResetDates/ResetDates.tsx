import React, {useRef} from 'react'
import styled from 'styled-components'
import RedoIcon from '../../icons/RedoIcon'
import Text from '../Text'
// eslint-disable-next-line import/no-unresolved
import {ResetDatesTheme} from '../../@types/theme'
import useThemeProps from '../../hooks/useThemeProps'
import globalStyles from '../../globalStyles'

const StyledReactDates = styled('button')`
  display: flex;
  align-items: center;
  cursor: pointer;
  border: 0;
  background: transparent;
  padding: 0;

  &:hover {
    text-decoration: underline;
  }
`

interface onResetDatesProps {
  onResetDates(): void
  text: string
}

function ResetDates({onResetDates, text}: onResetDatesProps) {
  const ref = useRef<HTMLButtonElement>(null)
  const theme: ResetDatesTheme = useThemeProps({
    fontFamily: globalStyles.fontFamily,
    resetDatesIconColor: globalStyles.colors.mud,
    resetDatesIconHeight: '14px',
    resetDatesIconWidth: '14px',
    resetDatesTextColor: globalStyles.colors.darcula,
    resetDatesTextMargin: '1px 0 0 8px',
    resetDatesTextLineHeight: 1.18,
    resetDatesTextFontSize: '11px',
  })

  function handleMouseUp() {
    if (ref && ref.current) {
      ref.current.blur()
    }
  }

  return (
    <StyledReactDates onClick={onResetDates} onMouseUp={handleMouseUp} ref={ref}>
      <RedoIcon
        // @ts-ignore
        height={theme.resetDatesIconHeight}
        // @ts-ignore
        width={theme.resetDatesIconWidth}
        // @ts-ignore
        color={theme.resetDatesIconColor}
      />
      <Text
        m={theme.resetDatesTextMargin}
        lineHeight={theme.resetDatesTextLineHeight}
        fontFamily={theme.fontFamily}
        fontSize={theme.resetDatesTextFontSize}
        // @ts-ignore
        color={theme.resetDatesTextColor}
      >
        {text}
      </Text>
    </StyledReactDates>
  )
}

export default ResetDates
