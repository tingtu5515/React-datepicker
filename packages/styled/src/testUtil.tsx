import React from 'react'
import {render} from '@testing-library/react'
import DatepickerContext, {datepickerContextDefaultValue} from './context/datepickerContext'

function renderWithProviders(ui: any, options?: any, contextValues?: Record<string, unknown>) {
  return render(
    <DatepickerContext.Provider value={{...datepickerContextDefaultValue, ...contextValues}}>
      {ui}
    </DatepickerContext.Provider>,
    options,
  )
}

export * from '@testing-library/react'
export {renderWithProviders as render}
