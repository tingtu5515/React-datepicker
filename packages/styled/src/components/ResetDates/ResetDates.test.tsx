import * as React from 'react'
import {render, fireEvent} from '../../testUtil'
import ResetDates from '.'

test('should match snapshot', () => {
  const {container} = render(<ResetDates rtl={false} text="Reset dates" onResetDates={jest.fn()} />)
  expect(container).toMatchSnapshot()
})

test('should render rtl variant', () => {
  const {container} = render(<ResetDates rtl text="Reset dates" onResetDates={jest.fn()} />)
  expect(container).toMatchSnapshot()
})

test('should execute onClick callback', () => {
  const onResetDates = jest.fn()
  const {container} = render(
    <ResetDates rtl={false} text="Reset dates" onResetDates={onResetDates} />,
  )
  // @ts-ignore
  fireEvent.click(container.firstChild)
  expect(onResetDates).toHaveBeenCalled()
  // @ts-ignore
  fireEvent.mouseUp(container.firstChild)
})
