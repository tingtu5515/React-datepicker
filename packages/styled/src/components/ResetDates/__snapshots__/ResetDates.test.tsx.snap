// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c1 {
  font-family: Montserrat,sans-serif;
  font-size: 11px;
  color: #343132;
  line-height: 1.18;
  margin: 1px 0 0 8px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  border: 0;
  background: transparent;
  padding: 0;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
  <button
    class="c0"
  >
    <svg
      class=""
      color="#58595B"
      height="14px"
      viewBox="0 0 14 14"
      width="14px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.015 11.15c-.027-.18-.04-.39-.067-.585a3.958 3.958 0 0 1-4.48-.056C2.663 9.241 2.142 6.663 3.292 4.74c1.217-2.02 3.797-2.592 5.696-1.282.589.404 1.03.934 1.35 1.533l-1.216.808L13 7.917l-.174-4.556-1.056.696a5.812 5.812 0 0 0-1.846-2.062C7.25.155 3.64.935 1.901 3.765c-1.672 2.717-.95 6.382 1.605 8.194a5.535 5.535 0 0 0 5.616.501c0-.083 0-.167-.013-.264a9.193 9.193 0 0 0-.094-1.046z"
        fill="currentColor"
        fill-rule="nonzero"
      />
    </svg>
    <div
      class="c1"
      color="#343132"
      font-family="Montserrat, sans-serif"
      font-size="11px"
    >
      Reset dates
    </div>
  </button>
</div>
`;

exports[`should render rtl variant 1`] = `
.c2 {
  font-family: Montserrat,sans-serif;
  font-size: 11px;
  color: #343132;
  line-height: 1.18;
  margin: 1px 8px 0 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  border: 0;
  background: transparent;
  padding: 0;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1 {
  -webkit-transform: rotate(-180deg);
  -ms-transform: rotate(-180deg);
  transform: rotate(-180deg);
}

<div>
  <button
    class="c0"
  >
    <svg
      class="c1"
      color="#58595B"
      height="14px"
      viewBox="0 0 14 14"
      width="14px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.015 11.15c-.027-.18-.04-.39-.067-.585a3.958 3.958 0 0 1-4.48-.056C2.663 9.241 2.142 6.663 3.292 4.74c1.217-2.02 3.797-2.592 5.696-1.282.589.404 1.03.934 1.35 1.533l-1.216.808L13 7.917l-.174-4.556-1.056.696a5.812 5.812 0 0 0-1.846-2.062C7.25.155 3.64.935 1.901 3.765c-1.672 2.717-.95 6.382 1.605 8.194a5.535 5.535 0 0 0 5.616.501c0-.083 0-.167-.013-.264a9.193 9.193 0 0 0-.094-1.046z"
        fill="currentColor"
        fill-rule="nonzero"
      />
    </svg>
    <div
      class="c2"
      color="#343132"
      font-family="Montserrat, sans-serif"
      font-size="11px"
    >
      Reset dates
    </div>
  </button>
</div>
`;
